src/transport/helpers/store-transport.js

```diff
@@ -0,0 +1,23 @@
+"use strict";
+
+/**
+ * Stores transport, adds listeners
+ */
+function storeTransport(transporter, transport, onCompleteCallback) {
+  // subscribe to disconnect/reconnect callbacks for transport
+  if (!transport) return onCompleteCallback(new Error("Unable to connect to an Ethereum node via any transport. (Web3, HTTP, WS, IPC)."), null);
+  transporter.internalState.transport = transport;
+  transport.addReconnectListener(function () {
+    Object.keys(transporter.internalState.reconnectListeners).forEach(function (key) {
+      transporter.internalState.reconnectListeners[key]();
+    });
+  });
+  transport.addDisconnectListener(function () {
```

Comment:
I'm nearly positive that the callback for the disconnect listener should (or at least, _can_, depending on transport) have an `event` parameter, which should be passed to the individual listeners:
```javascript
transporter.internalState.disconnectListeners[key](event);
```